Adds a section about Boyer-Moore's algorithm

LukasGamard opened the pull request:
First use of noweb syntax. I usually write my .tex files on Overleaf so I couldn't compile and double check.
Hope it is fine :)

Although [this wikipage](https://en.wikipedia.org/wiki/Boyer%E2%80%93Moore_string-search_algorithm) was a great source of information, I did not add it to the .bib file, nor did I quote it directly. Thought it would look a bit lower tier with wikipedia in the reference section...

LukasGamard committed:
Adds a section about Boyer-Moore's algorithm

dbosk commented on tutorials/search/contents.nw:
```diff
@@ -201,3 +201,64 @@ Det ger följande resultat.
+I vår enkel implementering inspirerad från~\cite{Sedgewick} måste vi fortfarande kolla på ett speciellfall. Om första tillfället av bokstaven vid en mismatch befinner sig till höger till mismatcha tecknet tillåter vi inte en hopp åt fel hålet. Då ska vi istället skifta mönstret ett steg till höger. Notera att detta kan vi behandla på samma sätt som tecken som inte finns i pattern, eftersom allt i ...
+
+\begin{frame}[fragile]
+<<använder next-array för att göra så stora hopp som möjligt>>
```
Saknar `=` i slutet av raden.

dbosk commented on tutorials/search/contents.nw:
```diff
@@ -201,3 +201,64 @@ Det ger följande resultat.
+Ni har säkert märkt att vi använder en \(pattern\) instans variabel. Låt oss se i detalj vad som händer inom konstruktorn.
+
+\begin{frame}[fragile]
+<<preprocessing pattern>>
```
Saknar `=` i slutet av raden.

dbosk commented on tutorials/search/bibliography.bib:
```diff
@@ -17,3 +17,32 @@ @techreport{RabinFingerprinting
+  title={Algorithms, fourth edition},
+  year={2011},
+  publisher={Pearson Education},
+  pages{770-773}
```
Missing `=`.

dbosk commented on tutorials/search/contents.nw:
```diff
@@ -201,3 +201,64 @@ Det ger följande resultat.
+
+\section{Boyer-Moore}
+
+Idéen bakom Boyer-Moores algoritmen är att göra större hopp igenom texten än med en brute-force metod genom att jämföra den sökta textsträngen~[[pattern]] med en lämgre textsträng~[[text]] från höger till vänster. Ett viktigt steg i metoden är att bestämma en KMP-automat innan vi börjar för att kunna avgöra i \(O(1)\) det största steget framåt vi får ta vid en missmatch. Detta preprocessing kan o ...
```
Stavfel i referens.

dbosk reviewed (commented):
Just these minor errors to make it compile. I haven't looked at the content yet. Will do soon.

LukasGamard committed:
Fixes some typos and does some rephrasing

dbosk committed:
Adds bm.py to Makefile, gitignore

dbosk committed:
Adds Boyer--Moore to abstract

dbosk committed:
Edits Boyer--Moore

- Fixes typos
- Minor edits
- Changes to disposition
- Adds complexity discussion

dbosk committed:
Adds Lukas to author list

dbosk committed:
Make all parts of Boyer--Moore source code use English

To be consistent with Rabin--Karp code.

dbosk committed:
Merge branch 'BoyerMoore' of github.com:LukasGamard/tilda into BoyerMoore

dbosk commented:
Great work @LukasGamard, thanks a lot!
